go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1771,7 +1771,20 @@ func flowsForDefaultBridge(bridge *bridgeConfiguration, extraIPs []net.IP) ([]st
 			if netConfig.isDefaultNetwork() {
 				continue
 			}
-			srcIPOrSubnet := netConfig.v4MasqIPs.ManagementPort.IP.String()
+			// The following flow looks like this:
+			//  cookie=0xdeff105, table=2, n_packets=0, n_bytes=0, priority=200,ip,nw_src=169.254.0.12 actions=mod_dl_dst:ce:c1:a6:6d:38:f1,output:3
+			// this flow matches on the specific UDN masIP and sends it to the corresponding UDN's patch port
+			// this is also what ensures isolation stays intact when UDNs try to access services in other UDNs or default networks
+			// side-note: for l3 networks its these flows that guarantee udn pod to different network's services isolation for nodeports but
+			// in l2 networks the fact that for same nodeport service in different network, the destination IP it the same as on the GR, the drop happens in OVN itself
+			// and we don't rely on these flows to achieve the isolation.
+			// These flows are also added for advertised networks because pod->nodePort traffic is SNATed to masqIP in case of advertised networks
+			// and hence these flows must be in place for consistency.
+			dftFlows = append(dftFlows,
+				fmt.Sprintf("cookie=%s, priority=200, table=2, ip, ip_src=%s, "+
+					"actions=set_field:%s->eth_dst,output:%s",
+					defaultOpenFlowCookie, netConfig.v4MasqIPs.ManagementPort.IP.String(),
+					bridgeMacAddress, netConfig.ofPortPatch))
```

Comment:
I would probably keep this flow, and the pkt mark flow close together in the code as before, the split confused me.